### PR TITLE
Refactor blob SAS tests and queue SAS test

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/test/ut/blob_sas_test.cpp [hunk 1]
```diff
@@ -317,301 +217,535 @@ namespace Azure { namespace Storage { namespace Test {
...
+
+    auto blobClient = *m_blockBlobClient;
+
+    Sas::AccountSasBuilder accountSasBuilder;
+    accountSasBuilder.Protocol = Sas::SasProtocol::HttpsAndHttp;
+    accountSasBuilder.ExpiresOn = sasExpiresOn;
+    accountSasBuilder.Services = Sas::AccountSasServices::Blobs;
+    accountSasBuilder.ResourceTypes = Sas::AccountSasResource::All;
+    accountSasBuilder.SetPermissions(Sas::AccountSasPermissions::All);
+    accountSasBuilder.ResourceTypes = Sas::AccountSasResource::Service;
```
Comment (microzchang) [inline sdk/storage/azure-storage-blobs/test/ut/blob_sas_test.cpp:578]: Why is `accountSasBuilder.ResourceTypes` assigned twice here?

**Review by microzchang — APPROVED** [review]

File: sdk/storage/azure-storage-blobs/test/ut/blob_sas_test.cpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/test/ut/blob_sas_test.cpp]: fixed.

**Pull request merged**